clients/da-vinci-client/src/main/java/com/linkedin/davinci/blobtransfer/server/P2PFileTransferServerHandler.java

```diff
@@ -172,6 +198,51 @@ public void exceptionCaught(ChannelHandlerContext ctx, Throwable cause) throws E
     ctx.close();
   }
 
+  /**
+   * Cleanup the snapshot directory if the snapshot is new generated for hybrid mode
+   * Decrease the user count for the snapshot to allow other hosts can use it.
+   * @param newSnapshotGenerated whether the snapshot is generated for hybrid mode
+   * @param snapshotDir the snapshot directory
+   * @param blobTransferRequest the blob transfer request
+   */
+  private void cleanupAndRemoveConcurrentUserRestriction(
```

Comment:
I haven't checked the method correctness in terms of the implementation but I think it's better to have this implemented in the blobSnapshotManager so it has some central locks to verify a single partition won't generate snapshots too frequently.